symfony/client/src/orangehrmAdminPlugin/pages/location/LocationList.vue

```diff
@@ -0,0 +1,317 @@
+                :options="countries"
+              />
+            </oxd-grid-item>
+          </oxd-grid>
+        </oxd-form-row>
+
+        <oxd-divider />
+
+        <oxd-form-actions>
+          <oxd-button displayType="ghost" label="Reset" @click="onClickReset" />
+          <oxd-button
+            class="orangehrm-left-space"
+            displayType="secondary"
+            label="Search"
+            type="submit"
+          />
+        </oxd-form-actions>
+      </oxd-form>
+    </oxd-table-filter>
+
+    <br />
+
+    <div class="orangehrm-paper-container">
+      <div class="orangehrm-header-container">
+        <div>
+          <oxd-button
+            label="Add"
+            iconName="plus"
+            displayType="secondary"
+            @click="onClickAdd"
+            v-if="$can.create(`locations`)"
+          />
+        </div>
+      </div>
+      <table-header
+        :selected="checkedItems.length"
+        :total="total"
+        :loading="isLoading"
+        @delete="onClickDeleteSelected"
+      ></table-header>
+      <div class="orangehrm-container">
+        <oxd-card-table
+          :headers="headers"
+          :items="items?.data"
+          :selectable="$can.delete(`locations`)"
+          :clickable="false"
+          :loading="isLoading"
+          v-model:selected="checkedItems"
+          v-model:order="sortDefinition"
+          rowDecorator="oxd-table-decorator-card"
+        />
+      </div>
+      <div class="orangehrm-bottom-container">
+        <oxd-pagination
+          v-if="showPaginator"
+          :length="pages"
+          v-model:current="currentPage"
+        />
+      </div>
+    </div>
+
+    <delete-confirmation ref="deleteDialog"></delete-confirmation>
+  </div>
+</template>
+
+<script>
+import usePaginate from '@orangehrm/core/util/composable/usePaginate';
+import {navigate} from '@orangehrm/core/util/helper/navigation';
+import {APIService} from '@/core/util/services/api.service';
+import DeleteConfirmationDialog from '@orangehrm/components/dialogs/DeleteConfirmationDialog.vue';
+import {computed, ref} from 'vue';
+import useSort from '@orangehrm/core/util/composable/useSort';
+
+const defaultFilters = {
+  name: '',
+  city: '',
+  countryCode: [],
+};
+
+const defaultSortOrder = {
+  'location.name': 'ASC',
+  'location.city': 'DEFAULT',
+  'country.countryCode': 'DEFAULT',
+  'location.phone': 'DEFAULT',
+  noOfEmployees: 'DEFAULT',
+};
+
+const locationDataNormalizer = data => {
+  return data.map(location => {
+    return {
+      id: location.id,
+      name: location.name,
+      city: location.city,
+      country: location.country.countryName,
+      phone: location.phone,
+      noOfEmployees: location.noOfEmployees ? location.noOfEmployees : 0,
+    };
+  });
+};
+
+export default {
+  props: {
+    countries: {
+      type: Array,
+      default: () => [],
+    },
+  },
+  data() {
+    const cellConfig = {};
+
+    if (this.$can.delete('locations')) {
```

Comment:
I think we should not do this check in data method, we should keep it as plain as possible since its a synchronous function running on component creation. also we have identified the need to have individual control over actions, we might add it to oxd lib later. for now we can use a computed property or since we have `:disabled` prop for `oxd-card-table` we can use it to disable all actions of table. I think `:disabled` prop should be fine since this is an admin screen.

```
<oxd-card-table
 :headers="headers"
 :items="items?.data"
 :selectable="$can.delete(`locations`)"
 :disabled="$can.update('locations')"
 :clickable="false"
 :loading="isLoading"
 v-model:selected="checkedItems"
 v-model:order="sortDefinition"
 rowDecorator="oxd-table-decorator-card"
/>
```

Reply:
@Chamara: The delete option is available in two places in the list.

1. Through the select checkboxes (bulk)
2. Through the Delete icon in each row

If we go ahead with the your solution, for a user without 'delete' data group permission, the first option above will be disabled, but the second option (delete icon) will still be available.

Wouldn't this be an issue?

Reply:
You are correct. however due to this being an admin screen, this is exclusive to users with Admin role so the user will always have both edit/delete permissions together. `:disabled="$can.update('locations') && $can.delete('locations')"` would be more pedantic but unnecessary. (fyi - if the table is disabled, all the action icons are un clickable)

Reply:
That's true, since we do not have custom user roles we can stay with this behavior. I moved the permission check into the widget attribute.